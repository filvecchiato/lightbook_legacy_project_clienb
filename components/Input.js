import React from 'react';
import { Text, StyleSheet, TextInput } from 'react-native';

const Input = (props) => {
  return (
    <>
      <Text style={styles.text}>{props.label}:</Text>
      <TextInput
        style={styles.input}
        value={props.value}
        onChangeText={props.setAction}
        placeholder={props.placeholder}
        type={props.type}
      />
    </>
  );
};

const styles = StyleSheet.create({
  input: {
    backgroundColor: 'white',
    width: '80%',
    padding: 15,
    marginBottom: 20,
    borderRadius: 3,
  },
  text: {
    fontSize: 25,
    paddingBottom: 5,
    color: 'white',
  },
});

export default Input;
