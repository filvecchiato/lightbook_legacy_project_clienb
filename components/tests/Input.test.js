import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

const tree = renderer.create(<Input label={'test'} value={'test2'} />).toJSON();

test('renders correctly', () => {
  expect(tree).toMatchSnapshot();
});

test('Pass the correct title from props', () => {
  const title = tree[0];
  expect(title.children[0]).toBe('test');
});

test('Pass the correct value from props', () => {
  const input = tree[1];
  expect(input.props.value).toBe('test2');
});
