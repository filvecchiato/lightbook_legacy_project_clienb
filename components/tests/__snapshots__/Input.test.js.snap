// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "paddingBottom": 5,
      }
    }
  >
    test
    :
  </Text>,
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 3,
        "marginBottom": 20,
        "padding": 15,
        "width": "80%",
      }
    }
    underlineColorAndroid="transparent"
    value="test2"
  />,
]
`;
